### PR TITLE
deps: bump go-git

### LINKED ISSUE #1000: Like Git does, act should also ignore the entries in git/info/exclude as well as the ones in ./.gitignore
## System information

- Operating System: Arch Linux
- Architecture: x64 (64-bit)
- Apple M1: nope
- Docker version: Docker version 20.10.12, build e91ed5707e
- Docker image used in `act`: Digest: sha256:c8897bb05bc79498622123863d4459ce4a585fabc03ec35757b5e008ef6804dd; Status: Image is up to date for ghcr.io/catthehacker/ubuntu:act-latest
- `act` version: act version 0.2.25

## Expected behaviour

```shell
testproj % act pull_request                                                                                       
[Run tests on Pull Request or Push/Run tests] 🚀  Start image=ghcr.io/catthehacker/ubuntu:act-latest
[Run tests on Pull Request or Push/Run tests]   🐳  docker pull image=ghcr.io/catthehacker/ubuntu:act-latest platform= username= forcePull=false
[Run tests on Pull Request or Push/Run tests]   🐳  docker create image=ghcr.io/catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Run tests on Pull Request or Push/Run tests]   🐳  docker run image=ghcr.io/catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Run tests on Pull Request or Push/Run tests]   🐳  docker exec cmd=[mkdir -m 0777 -p /var/run/act] user=root workdir=
[Run tests on Pull Request or Push/Run tests]   🐳  docker cp src=/home/matt/Projects/testproj/. dst=/home/matt/Projects/testproj
[Run tests on Pull Request or Push/Run tests]   🐳  docker exec cmd=[mkdir -p /home/matt/Projects/testproj] user= workdir=
[...]
```

This is where the process stopped. I expected it would continue until the workflow was finished.

## Actual behaviour

I received this error, and the process errored out:

```
Error: open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
```

This file caused the process to stop because it's a broken symlink in a remapped /tmp directory that I was using for development.

I didn't expect that it would be trying to access that file, since it's in my `.git/info/exclude` file, which I expected to be included in the gitignore filters that are used by default to filter files being uploaded to the container.

## Workflow and/or repository

This is not applicable to this situation, since it never got to the workflow. To recreate, make a broken symlink in your project directory and ignore it with `.git/info/exclude`. Then try any workflow and I assume it will happen.

## `act` output

<details>
  <summary>Log</summary>

```none
DEBU[0000] Loading environment from /home/matt/Projects/testproj/.env 
DEBU[0000] Loading secrets from /home/matt/Projects/testproj/.secrets
DEBU[0000] Loading workflows from '/home/matt/Projects/testproj/.github/workflows'
DEBU[0000] Loading workflows recursively                
DEBU[0000] Found workflow 'tagged-release.yml' in '/home/matt/Projects/testproj/.github/workflows/tagged-release.yml'
DEBU[0000] Found workflow 'test.yml' in '/home/matt/Projects/testproj/.github/workflows/test.yml'
DEBU[0000] Reading workflow '/home/matt/Projects/testproj/.github/workflows/tagged-release.yml'
DEBU[0000] Correcting if statements '/home/matt/Projects/testproj/.github/workflows/tagged-release.yml'
DEBU[0000] Reading workflow '/home/matt/Projects/testproj/.github/workflows/test.yml'
DEBU[0000] Correcting if statements '/home/matt/Projects/testproj/.github/workflows/test.yml'
DEBU[0000] Planning event: pull_request                 
DEBU[0000] Loading slug from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] Loading revision from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] HEAD points to 'a8d08eb3f59d240281ecc1efcb43e6365f0f5496'
DEBU[0000] using github ref: refs/heads/work-karma-up   
DEBU[0000] context env => map[ACT:true]                 
DEBU[0000] context needs => map[]                       
[Run tests on Pull Request or Push/Run tests] 🚀  Start image=ghcr.io/catthehacker/ubuntu:act-latest
DEBU[0000] Loading slug from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] Loading revision from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] HEAD points to 'a8d08eb3f59d240281ecc1efcb43e6365f0f5496'
DEBU[0000] using github ref: refs/heads/work-karma-up   
[Run tests on Pull Request or Push/Run tests]   🐳  docker pull image=ghcr.io/catthehacker/ubuntu:act-latest platform= username= forcePull=false
[Run tests on Pull Request or Push/Run tests]   🐳  docker pull ghcr.io/catthehacker/ubuntu:act-latest
DEBU[0000] Image exists? true                           
[Run tests on Pull Request or Push/Run tests] Removed container: b4e35aaf99491f0a5aea90601d2b1741488b0d6a89701ff4d5094005539f3ccb
[Run tests on Pull Request or Push/Run tests]   🐳  docker volume rm act-Run-tests-on-Pull-Request-or-Push-Run-tests
[Run tests on Pull Request or Push/Run tests]   🐳  docker create image=ghcr.io/catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Run tests on Pull Request or Push/Run tests] Created container name=act-Run-tests-on-Pull-Request-or-Push-Run-tests id=3a515cc5ac97debce07bf2c172749afc4989ddf4ba0e5c05c585c234a9e5ff62 from image ghcr.io/catthehacker/ubuntu:act-latest (platform: )
[Run tests on Pull Request or Push/Run tests] ENV ==> [RUNNER_TOOL_CACHE=/opt/hostedtoolcache RUNNER_OS=Linux RUNNER_TEMP=/tmp]
[Run tests on Pull Request or Push/Run tests]   🐳  docker run image=ghcr.io/catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Run tests on Pull Request or Push/Run tests] Starting container: 3a515cc5ac97debce07bf2c172749afc4989ddf4ba0e5c05c585c234a9e5ff62
[Run tests on Pull Request or Push/Run tests] Started container: 3a515cc5ac97debce07bf2c172749afc4989ddf4ba0e5c05c585c234a9e5ff62
[Run tests on Pull Request or Push/Run tests]   🐳  docker exec cmd=[mkdir -m 0777 -p /var/run/act] user=root workdir=
[Run tests on Pull Request or Push/Run tests] Exec command '[mkdir -m 0777 -p /var/run/act]'
[Run tests on Pull Request or Push/Run tests] Working directory '/home/matt/Projects/testproj'
[Run tests on Pull Request or Push/Run tests]   🐳  docker cp src=/home/matt/Projects/testproj/. dst=/home/matt/Projects/testproj
[Run tests on Pull Request or Push/Run tests]   🐳  docker exec cmd=[mkdir -p /home/matt/Projects/testproj] user= workdir=
[Run tests on Pull Request or Push/Run tests] Exec command '[mkdir -p /home/matt/Projects/testproj]'
[Run tests on Pull Request or Push/Run tests] Working directory '/home/matt/Projects/testproj'
DEBU[0000] Writing tarball /tmp/act3709751762 from /home/matt/Projects/testproj/.
DEBU[0000] Stripping prefix:/home/matt/Projects/testproj/ src:/home/matt/Projects/testproj/.
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
DEBU[0000] Loading slug from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] Loading revision from git directory '/home/matt/Projects/testproj/.git'
DEBU[0000] Found revision: a8d08eb3f59d240281ecc1efcb43e6365f0f5496
DEBU[0000] HEAD points to 'a8d08eb3f59d240281ecc1efcb43e6365f0f5496'
DEBU[0000] using github ref: refs/heads/work-karma-up   
DEBU[0000] context env => map[ACT:true AGENT_TOOLSDIRECTORY:/opt/hostedtoolcache DEBIAN_FRONTEND:noninteractive DEPLOYMENT_BASEPATH:/opt/runner IMAGE_OS:ubuntu20 ImageOS:ubuntu20 LSB_OS_VERSION:2004 LSB_RELEASE:20.04 PATH:/opt/hostedtoolcache/node/14.19.0/x64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin RUNNER_USER:root RUN_TOOL_CACHE:/opt/hostedtoolcache USER:root]
DEBU[0000] context needs => map[]                       
Error: open /home/matt/Projects/testproj/.local/excluded/_tmp/.com.google.Chrome.0NN5Rf/SingletonCookie: no such file or directory
```

</details>

By the way, I just found this project and It's already saving me a bunch of time, thanks. :-)

### PR DESCRIPTION
**Pull request by catthehacker**
fixes #1000

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/docker/distribution v2.8.1+incompatible
 	github.com/docker/docker v20.10.12+incompatible
 	github.com/go-git/go-billy/v5 v5.3.1
-	github.com/go-git/go-git/v5 v5.4.2
+	github.com/go-git/go-git/v5 v5.4.3-0.20211101214716-934e99b85109
```
Comment (ZauberNerd) [inline go.mod:14]: Which commit is that now? Is it the latest master of go-git?

Comment (catthehacker) [reply go.mod:14]: https://github.com/go-git/go-git/commit/934e99b85109874d8a24f344786c30276715bde0

Comment (catthehacker) [reply go.mod:14]: are there any improvements in later commits that we would benefit from?

Comment (ZauberNerd) [reply go.mod:14]: Yeah, maybe later. But I think I would prefer not to use too many forks, because dependency updates won't come in then, right?

Comment (catthehacker) [reply go.mod:14]: We don't get dependency updates anyway until `go-git` creates new release

**Review by KnisterPeter — APPROVED** [review]

**Review by cplee — APPROVED** [review]

**Review by ChristopherHX — APPROVED** [review]

**Pull request merged**